Author: Garrett Cooper <yanefbsd@gmail.com>
Description: Incremented address properly, so all lines are shown completely.

--- bsdmainutils.orig/usr.bin/hexdump/display.c	2011-06-06 15:46:30.000000000 +0200
+++ bsdmainutils/usr.bin/hexdump/display.c	2011-06-06 15:39:43.000000000 +0200
@@ -263,18 +263,20 @@
 			if (need == blocksize)
 				return((u_char *)NULL);
 			/*
-			 * XXX bcmp() is not quite right in the presence
-			 * of multibyte characters.
+			 * We need to explicitly bump the address displayed even
+			 * if -v isn't specified, or the address doesn't
+			 * increment with repeated output...
+			 * Please see bin/118723 for more details.
 			 */
+			eaddress = address + nread;
 			if (vflag != ALL && 
 			    valid_save && 
-			    bcmp(curp, savp, nread) == 0) {
+			    memcmp(curp, savp, nread) == 0) {
 				if (vflag != DUP)
 					(void)printf("*\n");
 				return((u_char *)NULL);
 			}
 			bzero((char *)curp + nread, need);
-			eaddress = address + nread;
 			return(curp);
 		}
 		n = fread((char *)curp + nread, sizeof(u_char),
@@ -289,13 +291,9 @@
 		if (length != -1)
 			length -= n;
 		if (!(need -= n)) {
-			/*
-			 * XXX bcmp() is not quite right in the presence
-			 * of multibyte characters.
-			 */
 			if (vflag == ALL || vflag == FIRST ||
 			    valid_save == 0 ||
-			    bcmp(curp, savp, blocksize) != 0) {
+			    memcmp(curp, savp, blocksize) != 0) {
 				if (vflag == DUP || vflag == FIRST)
 					vflag = WAIT;
 				return(curp);
